app/models/setting/salt.rb

```diff
@@ -1,14 +1,13 @@
 class Setting::Salt < Setting
-  def self.load_defaults
-    return unless super
+  def self.default_settings
+    [
+      set('salt_namespace_pillars', N_("Namespace Foreman pillars under 'foreman' key"), false),
+      set('salt_hide_run_salt_button', N_("Hide the Run Salt state.highstate button on the host details page"), false)
+    ]
+  end
 
-    transaction do
-      [
-        set('salt_namespace_pillars', N_("Namespace Foreman pillars under 'foreman' key"), false),
-        set('salt_hide_run_salt_button', N_("Hide the Run Salt state.highstate button on the host details page"), false)
-      ].each { |s| self.create! s.update(:category => 'Setting::Salt') }
-      Setting['salt_hide_run_salt_button'] = true if ForemanSalt.with_remote_execution?
-    end
-    true
+  def self.load_defaults
+    Setting['salt_hide_run_salt_button'] = true if ForemanSalt.with_remote_execution?
```

Comment:
Didn't wanted to be intrusive, but is this really what we want?
The setting should be disabled when `with_remote_execution?` is true, right?

Reply:
Yeah, but like this, user can still set the setting on false later, right? Just it will be set to true on every bootup. That seems odd to me.